explorer/src/lib/services/__tests__/duskAPI.spec.js

```diff
@@ -359,21 +359,118 @@ describe("duskAPI", () => {
   });
 
   it("should expose a method to search for blocks and transactions", async () => {
-    fetchSpy.mockResolvedValueOnce(
-      makeOKResponse(mockData.apiSearchBlockResult)
+    const fakeHash1 = Array(64).fill(1).join("");
+    const fakeHash2 = Array(64).fill(2).join("");
+    const hashResult = {
+      block: {
+        header: {
+          hash: fakeHash1,
+        },
+      },
+    };
+    const heightResult = {
+      block: {
+        header: {
+          hash: fakeHash2,
+        },
+      },
+    };
+
+    fetchSpy
+      .mockResolvedValueOnce(makeOKResponse(hashResult))
+      .mockResolvedValueOnce(makeOKResponse(heightResult));
+
+    await expect(duskAPI.search(node, fakeHash1)).resolves.toStrictEqual(
+      transformSearchResult([hashResult, heightResult])
     );
 
-    const query = "some search string";
+    expect(fetchSpy).toHaveBeenCalledTimes(2);
+    expect(fetchSpy.mock.calls[0][0]).toBe(gqlExpectedURL);
+    expect(fetchSpy.mock.calls[0][1]).toMatchInlineSnapshot(`
+      {
+        "body": "{"data":"\\n    query($id: String!) {\\n      block(hash: $id) { header { hash } },\\n      tx(hash: $id) { id }\\n    }\\n  ","topic":"gql"}",
+        "headers": {
+          "Accept": "application/json",
+          "Accept-Charset": "utf-8",
+          "Content-Type": "application/json",
+          "Rusk-gqlvar-id": ""1111111111111111111111111111111111111111111111111111111111111111"",
+        },
+        "method": "POST",
+      }
+    `);
+    expect(fetchSpy.mock.calls[1][0]).toBe(gqlExpectedURL);
+    expect(fetchSpy.mock.calls[1][1]).toMatchInlineSnapshot(`
+      {
+        "body": "{"data":"\\n    query($height: Float!) { block(height: $height) { header { hash } } }\\n  ","topic":"gql"}",
+        "headers": {
+          "Accept": "application/json",
+          "Accept-Charset": "utf-8",
+          "Content-Type": "application/json",
+          "Rusk-gqlvar-height": "1.1111111111111112e+63",
```

Comment:
Small issue here, this should not get formatted

Reply:
Well, it's not formatted: it's just the JSON representation of that big number, as it can be interpreted both as an hash (string) and as a block height (number).
Not a real situation, anyway, was just to test the case.